packages/account/src/Sections/Security/DeactivateAccount/deactivate-account-reason.jsx

```diff
@@ -0,0 +1,455 @@
+import React from 'react';
+import { localize, Localize } from '@deriv/translations';
+import { Formik, Field } from 'formik';
+import { Checkbox, Button, Input, FormSubmitButton, Modal, Icon, Money, Loading } from '@deriv/components';
+import CurrencyUtils from '@deriv/shared/utils/currency';
+import { connect } from 'Stores/connect';
+import { WS } from 'Services/ws-methods';
+
+const initial_form = {
+    otherFinancialPriorities: false,
+    stopTrading: false,
+    notIntrested: false,
+    anotherTradingWebsite: false,
+    notUserFriendly: false,
+    isDifficult: false,
+    lackFeaturesFunctionality: false,
+    unsatisfactoryCs: false,
+    otherReasons: false,
+    otherTradingPlatforms: '',
+    doToImprove: '',
+};
+
+const preparingReason = (values) => {
+    const selected_reasons = selectedReasons(values)
+        .map((val) => val[0])
+        .toString();
+    return `${selected_reasons}, ${values.otherTradingPlatforms}, ${values.doToImprove}`;
+};
+
+const selectedReasons = (values) => {
+    return Object.entries(values).filter(
+        ([key, value]) => !['otherTradingPlatforms', 'doToImprove'].includes(key) && value
+    );
+};
+
+const validateFields = (values) => {
+    const error = {};
+    const selected_reason_count = selectedReasons(values).length;
+    if (selected_reason_count === 0 && values.otherTradingPlatforms.length === 0 && values.doToImprove.length === 0) {
+        error.empty_reason = 'please select at least one reason or share your opinion';
+    } else if (selected_reason_count > 0 && (values.otherTradingPlatforms + values.doToImprove).length === 0) {
```

Comment:
formik validateFields usually, consist of failing conditions and add it to the error objects,

so here noticed only have 2 error meesages, so i think you dont need the second else if, and on the third elif, you can move the logic outside the if, and then second elif will be `!new RegExp(regex_rule).test(final_value)` instead